File: _REPO/APRESS/gdiplus-programming-in-csharp-vb.net/CH_8/PrintEvents-vb/MyPrintController.vb
Option Strict On

Imports System
Imports System.IO
Imports System.Drawing
Imports System.Drawing.Printing

Public Class MyPrintController
  Inherits StandardPrintController

  Private lblEvents As Label

  Public Sub New(ByRef lbl As Label)
    MyBase.New()
    lblEvents = lbl
  End Sub

  Public Overrides Sub OnStartPrint(ByVal doc As PrintDocument, _
                                    ByVal e As PrintEventArgs)
    lblEvents.Text += "      PrintController: OnStartPrint" + vbCrLf
    MyBase.OnStartPrint(doc, e)
  End Sub

  Public Overrides Function OnStartPage(ByVal doc As PrintDocument, _
                                        ByVal e As PrintPageEventArgs) _
                                        As Graphics
    lblEvents.Text += "      PrintController: OnStartPage" + vbCrLf
    Return (MyBase.OnStartPage(doc, e))
  End Function

  Public Overrides Sub OnEndPage(ByVal doc As PrintDocument, _
                                 ByVal e As PrintPageEventArgs)
    lblEvents.Text += "      PrintController: OnEndPage" + vbCrLf
    MyBase.OnEndPage(doc, e)
  End Sub

  Public Overrides Sub OnEndPrint(ByVal doc As PrintDocument, _
                                  ByVal e As PrintEventArgs)
    lblEvents.Text += "      PrintController: OnEndPrint" + vbCrLf
    MyBase.OnEndPrint(doc, e)
  End Sub


End Class

File: _REPO/APRESS/gdiplus-programming-in-csharp-vb.net/CH_8/PrintEvents-c/MyPrintController.cs
using System;
using System.Windows.Forms;
using System.Drawing;
using System.Drawing.Printing;
using System.ComponentModel;

namespace PrintEvents_c
{
	public class MyPrintController : StandardPrintController
	{
    private Label lblEvents;
		public MyPrintController(ref Label lbl): base()
		{
      lblEvents = lbl;
    }

    public override void OnStartPrint(PrintDocument doc, PrintEventArgs e)
    {
      lblEvents.Text += "      PrintController: OnStartPrint\n";
      base.OnStartPrint(doc, e);
    }
    public override Graphics OnStartPage(PrintDocument doc, PrintPageEventArgs e)
    {
      lblEvents.Text += "      PrintController: OnStartPage\n";
      return( base.OnStartPage(doc, e) );
    }
    public override void OnEndPage(PrintDocument doc, PrintPageEventArgs e)
    {
      lblEvents.Text += "      PrintController: OnEndPage\n";
      base.OnEndPage(doc, e);
    }
    public override void OnEndPrint(PrintDocument doc, PrintEventArgs e)
    {
      lblEvents.Text += "      PrintController: OnEndPrint\n";
      base.OnEndPrint(doc, e);
    }
  }
}
